src/main/java/br/com/zup/library/emprestimo/EmprestimoService.java

```diff
@@ -0,0 +1,69 @@
+package br.com.zup.library.emprestimo;
+
+import br.com.zup.library.exemplar.Exemplar;
+import br.com.zup.library.exemplar.ExemplarRepository;
+import br.com.zup.library.livro.Livro;
+import br.com.zup.library.usuario.Usuario;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.transaction.Transactional;
+
+import static br.com.zup.library.exemplar.TipoCirculacao.LIVRE;
+import static br.com.zup.library.usuario.TipoUsuario.PADRAO;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+@Service
+public class EmprestimoService {
+
+    private final EmprestimoRepository emprestimoRepository;
+    private final ExemplarRepository exemplarRepository;
+
+    @Autowired
+    public EmprestimoService(EmprestimoRepository emprestimoRepository,
+                             ExemplarRepository exemplarRepository) {
+        this.emprestimoRepository = emprestimoRepository;
+        this.exemplarRepository = exemplarRepository;
+    }
+
+    @Transactional
+    public NovoEmprestimoResponse realizaEmprestimo(SolicitacaoEmprestimo request) {
+        var usuario = request.getUsuario();
+        var livro = request.getLivro();
+
+        var exemplar = buscaExemplar(livro, usuario);
+
+        validaNumeroMaximoEmprestimos(usuario);
+
+        var novoEmprestimo = toModel(request, exemplar, usuario);
+
+        emprestimoRepository.save(novoEmprestimo);
+        exemplarRepository.save(exemplar.alteraDisponibilidade());
+
+        return new NovoEmprestimoResponse(novoEmprestimo);
+    }
+
+    private void validaNumeroMaximoEmprestimos(Usuario usuario) {
+        if (PADRAO.equals(usuario.getTipoUsuario()) && emprestimoRepository.findAllByUsuario(usuario).size() >= 5) {
```

Comment:
o `.findAllByUsuario(usuario)` está carregando TODOS os emprestimos do usuario pra simplesmente fazer um `size()` ? Se daqui a 3 anos o usuário tiver 1k emprestimos? ou 10k?

outro detalhe, vc verifica todos os emprestimos independente de algum status? digo, até os devolvidos?

Reply:
Ainda não tem lógica de devolução. Não sei se vou precisar excluir os registros ou por alguma TAG pra informar quando for devolvido. Porém o findAll vou alterar.

Reply:
resolvido